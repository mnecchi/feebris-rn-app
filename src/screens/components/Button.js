import React from 'react';
import {TouchableHighlight, Text} from 'react-native';

const Button = ({
  onPress,
  buttonStyle,
  textStyle,
  loading,
  disabled,
  title,
}) => (
  <TouchableHighlight
    onPress={onPress}
    style={buttonStyle}
    disabled={loading || disabled}>
    <Text style={textStyle}>{loading ? 'Checking...' : title}</Text>
  </TouchableHighlight>
);

export default Button;
