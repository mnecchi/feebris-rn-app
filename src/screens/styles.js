import {StyleSheet} from 'react-native';

// Common Styles

const labelText = {fontSize: 30, fontWeight: 'bold', color: '#444'};
const fieldContainer = {
  justifyContent: 'center',
  alignItems: 'center',
  margin: 16,
  maxWidth: '80%',
};
const switchContainer = {
  ...fieldContainer,
  textAlignVertical: 'center',
};

const iconInputContainer = {
  flexDirection: 'row',
  marginTop: 8,
  alignItems: 'center',
  justifyContent: 'center',
};

const buttonContainer = {
  padding: 8,
  backgroundColor: 'rgba(60, 142, 183, 1)',
  borderRadius: 8,
  marginTop: 64,
  width: '80%',
  maxWidth: 600,
  justifyContent: 'center',
  alignItems: 'center',
};

const buttonText = {...labelText, color: 'white'};

/**
 * Temperature, cough and fever screens styles
 */
const homeStyle = StyleSheet.create({
  container: {
    flex: 1,
    height: '100%',
    alignItems: 'center',
  },
  // TEMPERATURE
  temperatureContainer: fieldContainer,
  temperatureText: labelText,
  temperatureInput: {
    borderColor: '#ccc',
    borderWidth: 1,
    borderRadius: 8,
    padding: 8,
    fontSize: 20,
    width: 100,
    margin: 8,
  },
  temperatureInputError: {
    borderColor: 'red',
  },
  temperatureInputContainer: iconInputContainer,
  // COUGH
  coughContainer: switchContainer,
  coughText: {...labelText, paddingRight: 8},
  coughSwitchContainer: iconInputContainer,
  // FEVER IN LAST 5 DAYS
  feverContainer: switchContainer,
  feverText: {...labelText, paddingRight: 8},
  feverSwitchContainer: iconInputContainer,
  // BUTTON
  buttonContainer,
  buttonText,
});

/**
 * Results screen styles
 */
const resultsStyle = StyleSheet.create({
  container: {flex: 1, flexDirection: 'column'},
  topContainer: {
    alignItems: 'center',
    paddingBottom: 32,
    borderColor: '#ccc',
    borderWidth: 1,
  },
  // FLU CHECK RESULT
  fluResultContainer: {
    alignItems: 'center',
    marginTop: 32,
    marginHorizontal: 16,
  },
  fluResultText: {
    ...labelText,
  },
  fluResultFlu: {
    color: 'red',
  },
  fluResultNoFlu: {
    color: 'green',
  },
  // RESULTS TABLE
  tableContainer: {
    marginTop: 16,
    marginHorizontal: 8,
  },
  readingsTable: {
    borderWidth: 1,
    borderColor: '#ccc',
  },
  readingsTableHeader: {
    backgroundColor: 'rgba(60, 142, 183, 0.5)',
  },
  readingTableCell: {
    margin: 8,
    textAlign: 'center',
  },
  // BUTTON
  buttonContainer,
  buttonText,
});

export {homeStyle, resultsStyle};
