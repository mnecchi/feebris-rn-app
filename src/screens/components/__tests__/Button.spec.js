import React from 'react';
import 'react-native';
import Button from '../Button';
import renderer from 'react-test-renderer';
import {homeStyle} from '../../styles';

describe('should render successfully', () => {
  it('enabled and not loading', () => {
    const tree = renderer
      .create(
        <Button
          title="Title"
          onPress={jest.fn()}
          buttonStyle={homeStyle.buttonContainer}
          textStyle={homeStyle.buttonText}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('disabled', () => {
    const tree = renderer
      .create(
        <Button
          title="Title"
          disabled={true}
          onPress={jest.fn()}
          buttonStyle={homeStyle.buttonContainer}
          textStyle={homeStyle.buttonText}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('loading', () => {
    const tree = renderer
      .create(
        <Button
          title="Title"
          loading={true}
          onPress={jest.fn()}
          buttonStyle={homeStyle.buttonContainer}
          textStyle={homeStyle.buttonText}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
