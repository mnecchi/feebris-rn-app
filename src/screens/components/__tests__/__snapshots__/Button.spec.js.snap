// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render successfully disabled 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(60, 142, 183, 1)",
      "borderRadius": 8,
      "justifyContent": "center",
      "marginTop": 64,
      "maxWidth": 600,
      "padding": 8,
      "width": "80%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 30,
        "fontWeight": "bold",
      }
    }
  >
    Title
  </Text>
</View>
`;

exports[`should render successfully enabled and not loading 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(60, 142, 183, 1)",
      "borderRadius": 8,
      "justifyContent": "center",
      "marginTop": 64,
      "maxWidth": 600,
      "padding": 8,
      "width": "80%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 30,
        "fontWeight": "bold",
      }
    }
  >
    Title
  </Text>
</View>
`;

exports[`should render successfully loading 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(60, 142, 183, 1)",
      "borderRadius": 8,
      "justifyContent": "center",
      "marginTop": 64,
      "maxWidth": 600,
      "padding": 8,
      "width": "80%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 30,
        "fontWeight": "bold",
      }
    }
  >
    Checking...
  </Text>
</View>
`;
